cedar-policy-core/src/ast/entity.rs

```diff
@@ -38,56 +38,56 @@ pub static ACTION_ENTITY_TYPE: &str = "Action";
 #[derive(Serialize, Deserialize, PartialEq, Eq, Debug, Clone, Hash, PartialOrd, Ord)]
 #[cfg_attr(feature = "arbitrary", derive(arbitrary::Arbitrary))]
 #[serde(transparent)]
-pub struct EntityType(Name);
+pub struct EntityType(UnreservedName);
 
 impl EntityType {
     /// Is this an Action entity type?
     /// Returns true when an entity type is an action entity type. This compares the
     /// base name for the type, so this will return true for any entity type named
     /// `Action` regardless of namespaces.
     pub fn is_action(&self) -> bool {
-        self.0.basename() == &Id::new_unchecked(ACTION_ENTITY_TYPE)
+        self.0.as_ref().basename() == &Id::new_unchecked(ACTION_ENTITY_TYPE)
     }
 
     /// The name of this entity type
-    pub fn name(&self) -> &Name {
+    pub fn name(&self) -> &UnreservedName {
         &self.0
     }
```

Comment:
Curious why change the return type here, rather than keep the signature of this function the same and make the implementation `self.as_ref()`

Reply:
It's addressed after the renaming?

Reply:
Yes